### DISCUSSION TITLE
Long polling locks the thread

### DISCUSSION THREAD

**Discussion by MrVolot** [reo7sp/tgbot-cpp#261] [Q&A] [answered]
`start()` of the `TgLongPoll` class locks the main thread after some time of working (several hours). I know the `start()` should be called in the infinite loop. The problem is that there are no errors or exceptions. The bot just stops working, since no updates are processed. Is there any way to handle it and is it a known workflow or a bug?

**Comment by reo7sp** [comment, marked as answer]
The old HTTP implementation from this report was replaced in v1.12. Current libcurl requests have connection and total timeouts, and `TgLongPoll::startLoop(errorHandler)` handles transient polling errors without a manual infinite loop.
